Source/MLX/GPU.swift

```diff
@@ -21,6 +21,8 @@ import Foundation
 /// - ``snapshot()``
 public enum GPU {
 
+    static let queue = DispatchQueue(label: "GPUEnum")
```

Comment:
Use this to guard the mutable static properties.  I didn't mark any of this as sendable (we are just using the enum like a namespace) but we can add it later if needed.  This is thread safe now.